### PR TITLE
Add lint workflow

### PR DESCRIPTION
**Pull request by mccalluc**
This is relevant to my project of separating dependencies, because if a library import is unused, it introduces a dependency where it's not really necessary.

Making a minimal PR now that just introduces linting, and then I'll branch from this and fix some of the things it identifies.

### CODE REVIEW

**Comment by keller-mark** [comment]
Thanks, this is great. Can we also add `flake8` to the main `environment.yml`, and a linting section of the README explaining how to run the linting locally when developing the package?

**Comment by mccalluc** [comment]
@keller-mark -- Can you formally review and approve if this looks good?

**Comment by keller-mark** [comment]
If I run `flake8` in the root directory of the repo, what is the expected output?

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,27 @@
 [bdist_wheel]
 universal=1
+
+[flake8]
+exclude = js/node_modules
```
Comment (keller-mark) [inline setup.cfg:5]:
```suggestion
```
Apologies, the most recent PR into this branch also added an exclude section

**Pull request merged**